app/code/Magento/Config/Model/Config/Source/Locale/Currency.php

```diff
@@ -41,7 +56,36 @@ public function toOptionArray()
         if (!$this->_options) {
             $this->_options = $this->_localeLists->getOptionCurrencies();
         }
-        $options = $this->_options;
+        $options = [];
+        $selected = $this->_getInstalledCurrencies();
+        
+        foreach ($this->_options as $option) {
```

Comment:
Instead of using the `foreach`, consider using the `array_filter()` method:

```
$options = array_filter(
  $this->_options,
  function ($option) use ($selected) {
    return in_array($option, $selected);
  }
);
```

Reply:
Ok, if you believe it will gain anything

Reply:
done